src/ui/customBreakpointsUI.ts

```diff
@@ -125,27 +174,67 @@ class BreakpointsDataProvider implements vscode.TreeDataProvider<Breakpoint | Ca
       breakpoint.checkboxState = state;
 
       if (breakpoint instanceof Category) {
-        for (const child of breakpoint.children) {
+        for (const child of this.getChildren(breakpoint) as XHRBreakpoint[]) {
           if (child.checkboxState !== state) {
             child.checkboxState = state;
           }
         }
-      } else {
-        if (!enabled && breakpoint.parent.checked) {
-          breakpoint.parent.checkboxState = state;
-        } else if (enabled && breakpoint.parent.children.every(c => c.checked)) {
-          breakpoint.parent.checkboxState = state;
+      } else if (breakpoint instanceof Breakpoint || breakpoint instanceof XHRBreakpoint) {
+        const parent = this.getParent(breakpoint) as Category;
+        if (!enabled && parent.checked) {
+          parent.checkboxState = state;
+        } else if (enabled && parent.children.every(c => c.checked)) {
+          parent.checkboxState = state;
         }
       }
     }
 
     const ids = this.allBreakpoints.filter(b => b.checked).map(b => b.id);
+    const xhr = this.xhrBreakpoints.filter(b => b.checkboxState).map(b => b.id);
+    const xhrDisabled = this.xhrBreakpoints.filter(b => !b.checkboxState).map(b => b.id);
     for (const session of this._debugSessionTracker.getConcreteSessions()) {
       session.customRequest('setCustomBreakpoints', { ids });
+      session.customRequest('enableXHRBreakpoints', { ids: xhr });
+      session.customRequest('disableXHRBreakpoints', { ids: xhrDisabled });
```

Comment:
Perhaps we make `setCustomBreakpoints` take the `ids` as well as a list of `xhrPatterns: string[]` -- avoid having extra calls and work.